pkg/controllers/management/node/utils.go

```diff
@@ -67,8 +67,20 @@ func buildCreateCommand(node *v3.Node, configMap map[string]interface{}) []strin
 	cmd = append(cmd, buildEngineOpts("--engine-registry-mirror", node.Status.NodeTemplateSpec.EngineRegistryMirror)...)
 	cmd = append(cmd, buildEngineOpts("--engine-storage-driver", []string{node.Status.NodeTemplateSpec.EngineStorageDriver})...)
 
+	// Append driver-specific flags to the command.
+	cmd = append(cmd, buildDriverFlags(sDriver, configMap)...)
+
+	// Add the hostname to the command.
+	cmd = append(cmd, node.Spec.RequestedHostname)
+	return cmd
+}
+
+// buildDriverFlags extracts driver-specific configuration from the given configmap and turns it into CLI flags.
+func buildDriverFlags(driverName string, configMap map[string]interface{}) []string {
+	cmd := make([]string, 0)
```

Comment:
nit.

```suggestion
	flags := make([]string, len(configMap))
```

Reply:
Based on the code, I don't think `cmd` will necessarily have the same number of entries as the configmap, so this wouldn't work.

Reply:
Ok, it makes sense.